mono/mini/exceptions-arm64.c

```diff
@@ -620,3 +626,598 @@ mono_arch_do_ip_adjustment (MonoContext *ctx)
+	mono_arch_unwind_add_alloc_x (unwind_codes, unwind_code_size, offset, reverse);
+}
+
+typedef struct _InitilizedWindowInfoResult {
+	gint32 saved_int_regs;
+	gboolean can_use_packed_format;
+	gboolean needs_seperate_epilog;
+} InitilizedWindowInfoResult;
+
+static InitilizedWindowInfoResult
+initialize_unwind_info_prolog (GSList* unwind_ops, gint stack_offset, guint param_area, guint8 *unwind_codes, guint32 *unwind_code_size) {
+
+	InitilizedWindowInfoResult res;
+	res.saved_int_regs = 0;
+	res.needs_seperate_epilog = FALSE;
+	res.can_use_packed_format = TRUE;
+
+	if (enable_ptrauth)
+		mono_arch_unwind_add_pac_sign_lr (unwind_codes, unwind_code_size);
+
+	// Stack probes
+	if (stack_offset + param_area > 0x1000) {
+		//  stp    fp,lr,[sp,#-16]
+		//  ld imm	(1 or 2 instructions)
+		//  call __chkstk
+		//  ldp    fp,lr,[sp,#16]
+		guint probe_size = (stack_offset + param_area) / 16;
+		mono_arch_unwind_add_nops(unwind_codes, unwind_code_size, probe_size > 0xFFFF ? 5 : 4);
+		res.can_use_packed_format = FALSE;
+		res.needs_seperate_epilog = TRUE;
+	}
+
+	// Frame Setup
+	if (arm_is_ldpx_imm(-stack_offset)) {
+		if (stack_offset > 0)
+			mono_arch_unwind_add_save_fplr_x (unwind_codes, unwind_code_size, stack_offset);
+		else
+			mono_arch_unwind_add_save_fplr (unwind_codes, unwind_code_size, 0);
+	}
+	else {
+		mono_arch_unwind_add_emit_subx_sp_imm(stack_offset, unwind_codes, unwind_code_size, TRUE);
+		mono_arch_unwind_add_save_fplr (unwind_codes, unwind_code_size, 0);
+		res.can_use_packed_format = FALSE;
+	}
+
+	mono_arch_unwind_add_set_fp (unwind_codes, unwind_code_size);
+
+	if (param_area) {
+		mono_arch_unwind_add_emit_subx_sp_imm(param_area, unwind_codes, unwind_code_size, TRUE);
+		res.can_use_packed_format = FALSE;
+	}
+
+	gint32 last_saved_in_order_reg = -1;
+	guint32 last_saved_reg_offset = -1;
+	
+	MonoUnwindOp* unwind_op_data;
+
+	// Replay collected unwind info and setup Windows format.
+	for (GSList* l = unwind_ops; l; l = l->next) {
+		unwind_op_data = (MonoUnwindOp*)l->data;
+
+		switch (unwind_op_data->op) {
+		case DW_CFA_mono_prolog_nop_win64_arm64:
+			res.needs_seperate_epilog = TRUE;
+			mono_arch_unwind_add_nops (unwind_codes, unwind_code_size, unwind_op_data->val);
+			break;
+		case DW_CFA_mono_epilog_nop_win64_arm64:
+			res.needs_seperate_epilog = TRUE;
+			break;
+		case DW_CFA_mono_fp_alloc_info_win64_arm64:
+			mono_arch_unwind_add_alloc_x (unwind_codes, unwind_code_size, unwind_op_data->val, TRUE);
+			break;
+		case DW_CFA_mono_restore_offset_win64_arm64: 
+			res.needs_seperate_epilog = TRUE;
+			break;
+		case DW_CFA_offset: {
+			guint offset = stack_offset + unwind_op_data->val;
+			int reg_count;
+
+			reg_count = mono_arch_unwind_add_reg_offset(unwind_op_data, l->next, stack_offset, unwind_codes, unwind_code_size, TRUE);
+
+			if (reg_count > 0) {
+				if (last_saved_in_order_reg == -1)
+					res.can_use_packed_format = (unwind_op_data->reg == ARMREG_R19);
```

Comment:
is it possible for res.can_use_packed_format to already be false and the result of `(unwind_op_data->reg == ARMREG_R19)` will make it true?  Maybe `if (last_saved_in_order_reg == -1 && res.can_use_packed_format)`

Reply:
Fixed in https://github.com/Unity-Technologies/mono/pull/1960/commits/cec9c68347b719dfbccb1a07e1f859c903bbf4d5